crates/polars-plan/src/plans/ir/text_plan_graph/models.rs

```diff
@@ -0,0 +1,227 @@
+use polars_core::frame::UniqueKeepStrategy;
+use polars_ops::frame::{JoinCoalesce, JoinValidation, MaintainOrderJoin};
+use polars_utils::pl_str::PlSmallStr;
+use polars_utils::unique_id::UniqueId;
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "dsl-schema", derive(schemars::JsonSchema))]
+#[derive(Debug)]
+pub struct TextPlanGraph {
+    pub title: PlSmallStr,
+    /// Number of roots
+    pub roots: usize,
+    pub nodes: Vec<TpgNode>,
+    pub edges: Vec<Edge>,
+    pub legend: Option<Vec<PlSmallStr>>,
+}
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "dsl-schema", derive(schemars::JsonSchema))]
+#[derive(Debug, Default)]
+pub struct TpgNode {
+    pub id: u64,
+    pub title: PlSmallStr,
+    pub properties: IRTpgProperties,
+    pub tags: u128,
+}
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "dsl-schema", derive(schemars::JsonSchema))]
+#[derive(Debug, Default)]
+pub struct Edge {
+    pub source: u64,
+    pub target: u64,
+}
+
+impl Edge {
+    pub fn new<T, U>(source: T, target: U) -> Self
+    where
+        u64: TryFrom<T> + TryFrom<U>,
+        <u64 as TryFrom<T>>::Error: std::fmt::Debug,
+        <u64 as TryFrom<U>>::Error: std::fmt::Debug,
+    {
+        Self {
+            source: source.try_into().unwrap(),
+            target: target.try_into().unwrap(),
+        }
+    }
+}
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(tag = "type"))]
+#[cfg_attr(feature = "dsl-schema", derive(schemars::JsonSchema))]
+#[derive(Debug, Default, strum_macros::IntoStaticStr)]
+pub enum IRTpgProperties {
```

Comment:
I think this diverged quite a bit from the original idea of having a uniform interchange format for all the plans. I don't really see the point of this over just